### DISCUSSION TITLE
Naming conventions and dependencies

### DISCUSSION THREAD

**Discussion by rajesh3007** [Azure-Player/azure.datafactory.tools#284] [Q&A] [answered]
1. Does the target data factory have the same names as the source data factory for the following:

-  Linked services
-  Integrated runtime (we are using self-hosted IR)
-  global variables
-  pipeline parameters
- Triggers

Currently, we have different names for all the above-mentioned from source to target data factory and the data factories are in different resource groups but in the same subscription.

2. In the include/exclude filter do I have to include the dependencies list as well?
example: when promoting a pipeline (included in filter list) do I have to include the dependencies list as well? such as its linked services and IR? 

Let me know if you have further questions on this and I can provide you with more information. Any help is much appreciated.

Thank you.

**Comment by NowinskiK** [comment, marked as answer]
First of all - I'm not sure how you want to handle different names of objects between environments...
#adftools doesn't support renaming objects during the deployment. The only natural exception is ADF instance itself - ADF must have unique name across the Azure globally, so target Subscription or Resource Group doesn't matter.

Where do you have master code of ADF? GIT repository? 
If so - when & why are you changing names for LS, global variables, triggers, etc for other environments?

 ad.2)
Yes. If you want to use selective deployment, you should exactly know WHY you do that and which objects should be deploy (include) or not (exclude). Folders (ADF) or wildcards can help with that.
That's theory. As long as you know how it works - you can build your own approach for the edge case, for example:
a) one release pipeline for deploying INFRA + all Linked Services
b) another one for deploying selectively (ADF pipelines, datasets, etc)
You have full flexibility here.
BTW. SHIR should be deploy only once (it's INFRA), before any other ADF artifacts.

**Reply by rajesh3007** [reply]
It's my first time doing an ADF deployment, earlier we didn't have multiple ADF instances. 

We are using Azure DevOps as our repository for our DEV ADF, and we have not set up any repo for the target ADF instance. The goal is to make changes in Dev and push changes to target ADF.

We thought of implementing it with **"ARM template deployment on resource group"** (task in azure DevOps), this would have worked fine if we are doing one deployment per sprint but now, we are doing multiple deployments per sprint and only specific pipelines are to be deployment as other might be still in testing/development phase. So, we are trying to see our options:

1. use the extension (marketplace task) adftools --> we chose to go with this so that we can save some time
or
2. write the PowerShell scripts to do this job in house (time-consuming)


I appreciate your explanation; it gave us some insights on how we can plan our deployments using ADFtool (will test it for our use case)
